envoy/service/auth/v2/external_auth.proto

```diff
@@ -26,4 +26,22 @@ message CheckRequest {
 message CheckResponse {
   // Status `OK` allows the request. Any other status indicates the request should be denied.
   google.rpc.Status status = 1;
+
+  // An optional message which contains the attributes of an HTTP response object. This message is
+  // particularly useful when the authorization service needs to send specific responses to the
+  // downstream client or to modify the request headers before dispatching them to the upstream.
+  // E.g., when the authorization status is not OK, this message can be used by the authorization
+  // service to the tell the filter to respond with specific HTTP headers, body, and status code.
+  // When the authorization status is OK, this object may include the headers that should be
+  // modified in the original request before it gets sent to the upstream.
+  message HttpResponse {
+    // Http status code.
+    int32 status_code = 1;
```

Comment:
You probably want to make this consistent with other uses of status code in the API, e.g. https://github.com/envoyproxy/data-plane-api/blob/641920a96a4a379a959e080b2981aaa672073fa7/envoy/api/v2/route/route.proto#L582